src/App.tsx

```diff
@@ -263,7 +267,22 @@ export const App = ({
 		) {
 			onExpanded(window.performance.now() - expandedStartTime);
 		}
-	}, [isExpanded]);
+
+		if (isExpanded) {
+			// We want react to complete the current work and render, without trying to batch this update
+			// before resetting the number of comments
+			// to the total comment amount.
+			// This allows a quick render of minimal comments and then immediately begin rendering
+			// the remaining comments.
+			setLoadingMore(true);
+			const timer = setTimeout(() => {
```

Comment:
I always feel like this is a code smell but I don't think there's a way to force React not to batch the update (i.e without `setTimeout`, the `setNumberOfCommentsToShow` gets set *before* render, and therefore behaves in the exact same way as before as the state update gets batched)

Reply:
setImmediate`? I agree though, the fact it's needed is hinting at there's a better way but it's not screaming at me. My only thought is to have another component that deals with rendering the list and try to capture the flow inside that using useEffect. I wonder though if a simple setImmediate (or setTimeout) like yuoo've got here isn't simpler?

Not a blocker, just a passing thought, maybe a `useComments` custom hook might work well here to reduce this complexity? Maybe the api could look like:

```typescript
const {
  comments,
  loading,
  loadingMore,
  totalComments,
} = useComments(url, options);
```

Reply:
Just to check my understanding, this is to defer execution?

Reply:
> setImmediate

Only IE supports, but essentially, yea.

> I wonder though if a simple setImmediate (or setTimeout) like yuoo've got here isn't simpler?

Yeah, I feel like it. Took me a bit to walk through this file already, it's possibly worth refactoring if we implement similar elsewhere but for now seems simple enough.

> maybe a useComments custom hook might work well here to reduce this complexity

Yeah, that would help make App.tsx more manageable. I'll give it a go and follow up with another PR.

Reply:
> Just to check my understanding, this is to defer execution?

Yes, it pushes the execution of the callback to the back of the event queue, so React will complete it's update before executing it, and therefore not batch this `setState` update before render.